docs/design/integrate-with-cc-tools/README.md

```diff
@@ -0,0 +1,308 @@
+
+On the other hand, the complexity of developing, testing, and deploying smart contracts (chaincode) in a decentralized environment presents its challenges. The CC Tools project, a Hyperledger Lab, is designed to streamline this process by providing a simplified, user-friendly framework for chaincode development. It reduces the time and effort required for developers to develop and test chaincode.
+
+This project seeks to integrate the privacy guarantees of FPC with the developer-friendly environment of CC Tools to create a unified solution that offers enhanced security and ease of development. Doing so, not only improves the privacy model of smart contract execution but also simplifies the workflow for developers, making the benefits of FPC more accessible. This integration serves as a powerful tool for organizations seeking to develop secure and private decentralized applications efficiently, without compromising on usability or privacy.
+
+## PROBLEM STATEMENT
+
+This integration can't be done implicitly as both projects are not tailored to each other. CC-tools is created based on using standard fabric networks so it doesn't handle encryption while communicating between the ledger and peers. FPC also was never tested with CC-tools packages as both utilize fabric stub interface in their stub wrapper
+
+The integration between FPC and CC Tools needs to occur at two distinct levels:
+
+Chaincode Level: The chaincode itself will be written using CC Tools to simplify its complexity and must be compatible with the privacy features of FPC. This requires modifications to the chaincode lifecycle, ensuring that it can be deployed securely using FPC while retaining the flexibility of CC Tools for development.
+Note: Adhering to FPC's security requirements adds more complexity to the chaincode deployment process than with standard fabric
+
+Client-Side Level: The client that communicates with the peers must be adapted to handle the secure interaction between the FPC chaincode and the Fabric network. This includes managing secure communication with the TEE and ensuring that FPC's privacy guarantees are upheld during transaction processing.
+
+Addressing these challenges is critical to enabling the seamless development and deployment of private chaincode on Hyperledger Fabric, ensuring both security and usability.
+Note: the following list comprises the functionality supported by cc-tools but not yet by FPC
+
+* Transient data
+* Events
+* PrivateData
```

Comment:
I think this note should move to a different section, maybe limitations and future work?

Reply:
I talk about them already in the limitations and future work sections but I thought this can be a useful note in the problem statement, or?